src/map/lua/lua_baseentity.cpp

```diff
@@ -8734,8 +8734,8 @@ inline int32 CLuaBaseEntity::checkKillCredit(lua_State *L)
 
     bool credit = false;
     int lvlDiff = (int)(PMob->GetMLevel()) - (int)(PChar->GetMLevel());
-    int maxDiff = (lua_isnil(L, 2) || !lua_isnumber(L, 2)) ? (int)lua_tonumber(L, 2) : 15;
-    float range = (lua_isnil(L, 3) || !lua_isnumber(L, 3)) ? (float)lua_tonumber(L, 3) : 100;
+    int maxDiff = (!lua_isnil(L, 2) && lua_isnumber(L, 2)) ? (int)lua_tonumber(L, 2) : 15;
```

Comment:
Does `lua_isnumber` actually return true if the value is nil?

Reply:
Yes
**https://pgl.yoyo.org/luai/i/lua_isnumber**

Reply:
But.. but.. _nil_ isn't a string! :cry: